src/libstore/local-fs-store.cc

```diff
@@ -33,6 +33,10 @@ struct LocalStoreAccessor : PosixSourceAccessor
 
     std::optional<Stat> maybeLstat(const CanonPath & path) override
     {
+        /* Handle the case where `path` is (a parent of) the store. */
+        if (isDirOrInDir(store->storeDir, path.abs()))
+            return Stat{ .type = tDirectory };
```

Comment:
It's not clear why this is needed (calling it with a parent of the store feels quite wrong).

I guess it should at least honor `requireValidPath`

Reply:
Because when following a symlink from one store path to another, `resolveSymlinks()` will stat the parents of those paths, so it will do a `maybeLstat("/nix")` and `maybeLstat("/nix/store")`. Those are not valid store paths so it failed.

Reply:
I've been thinking about whether stores' source accessors should prefix everything with the store directory or not, and leaning towards "no".

This would be something not doing that helps with: the path is just `xxxxxxxxxxxx-foo`, no parent directories etc.

Reply:
`makeMountedInputAccessor` can be used to put put things in store directories if it matters.

Reply:
#10510 Broke into new issue

Reply:
Given that #10510 *isn't* fixed, I think this is actually a fine fix for now.